lib/membrane_h264_plugin/parser/nalu_parser.ex

```diff
@@ -15,34 +14,32 @@ defmodule Membrane.H264.Parser.NALuParser do
             scheme_parser_state: SchemeParser.t(),
             has_seen_keyframe?: boolean()
           }
-  @enforce_keys [:scheme_parser_state, :has_seen_keyframe?]
-  defstruct @enforce_keys
+
+  defstruct scheme_parser_state: SchemeParser.new(), has_seen_keyframe?: false
 
   @doc """
   Returns an structure holding a NALu parser state.
   """
   @spec new() :: t()
-  def new(),
-    do: %__MODULE__{
-      scheme_parser_state: SchemeParser.new(),
-      has_seen_keyframe?: false
-    }
+  def new(), do: %__MODULE__{}
 
   @doc """
-  Parses a `Membrane.Buffer.t()` with a payload being a part of H264 stream.
+  Parses a binary representing a single NALu.
 
-  As a result it returns a list of NAL units, along with a part of a payload
-  that wasn't completly parsed and the updated state of the NALu parser.
+  Returns a structure of type `Membrane.H264.Parser.NALu` that
+  contains parsed fields fetched from that NALu.
```

Comment:
Not sure if this is needed as we clearly indicate what is returned via typespec. However if we want to keep this you can refer to the type with `t:Membrane.H264.Parser.NALu.t/0`.